cmake/BundledOSSPUUID.cmake

```diff
@@ -21,8 +21,11 @@ function(use_bundled_osspuuid SOURCE_DIR BINARY_DIR)
     message("Using bundled ossp-uuid")
 
     # Define patch step
-    set(PC "${Patch_EXECUTABLE}" -p1 -i "${SOURCE_DIR}/thirdparty/ossp-uuid/ossp-uuid-mac-fix.patch" &&
-           "${Patch_EXECUTABLE}" -p1 -i "${SOURCE_DIR}/thirdparty/ossp-uuid/ossp-uuid-no-prog.patch")
+    set(PC "${Patch_EXECUTABLE}" -p1 -N -i "${SOURCE_DIR}/thirdparty/ossp-uuid/ossp-uuid-mac-fix.patch" &&
+            "${Patch_EXECUTABLE}" -p1 -N -i "${SOURCE_DIR}/thirdparty/ossp-uuid/ossp-uuid-no-prog.patch" ||
+            # if already applied, reverse application should succeed
+            "${Patch_EXECUTABLE}" -p1 -R --dry-run -i "${SOURCE_DIR}/thirdparty/ossp-uuid/ossp-uuid-mac-fix.patch" &&
+            "${Patch_EXECUTABLE}" -p1 -R --dry-run -i "${SOURCE_DIR}/thirdparty/ossp-uuid/ossp-uuid-no-prog.patch")
```

Comment:
It would be nicer to have a `( ... )` around the `--dry-run` commands.  Otherwise, if the first two patch commands succeed, we skip the third command, but execute the fourth.  This is not a big problem, as the fourth command should succeed, but it isn't nice.

Reply:
https://github.com/szaszm/nifi-minifi-cpp/commit/fee00580740c578a64956d3e448c7729f9bb44b0